File: Usuarios/src/main/java/dev/carloscastano/get/entities/Address.java
package dev.carloscastano.get.entities;

import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.*;

@Entity
@Table(name = "direccion")
@Setter
@Getter
@ToString
@EqualsAndHashCode
public class Address {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id_direccion;
    private String calle;
    private String ciudad;
    private String departamento;
    private String codigo_postal;
    private String tipo;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_usuario")
    @JsonBackReference
    private User usuario;
}

File: App/src/app/models/users/Address.ts
import { TypeAddress } from "./TypeAddress";
import { User } from "./User";

export interface Address {
    idDireccion: number;
    idUsuario: User["idUsuario"];
    calle: string;
    ciudad: string;
    departamento: string;
    codigoPostal: string;
    tipoDireccion: TypeAddress;
}